app/interactors/tasks/delete.rb

```diff
@@ -0,0 +1,13 @@
+module Tasks
+
+  class Delete
+    include Interactor
+
+    delegate :task, to: :context
+
+    def call
+      context.fail!(error: "Something went wrong, project cannot be deleted!") unless task.delete
```

Comment:
Don't use `delete` method because it is not run any callbacks, dependent too

```suggestion
      context.fail!(error: "Something went wrong, project cannot be deleted!") unless task.destroy
```